ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml

```diff
@@ -25,5 +25,7 @@ ref:
     env: OPENSHIFT_INSTALL_RELEASE_IMAGE
   - name: "release:latest"
     env: OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
+  - name: "agent-preinstall-image-builder"
+    env: APPLIANCE_IMAGE
```

Comment:
I'm not sure about this one, couldn't be simply set via DEVSCRIPTS_CONFIG? The step is shared by many different jobs (from different teams), and it should have only very generic env vars

Reply:
It should be done using dependencies as it translates the url (to the local image that is built in job). Couldn't find a more isolated way for it...

Reply:
As for many other workflows based on this step specific dev-scripts configuration must be placed inside DEVSCRIPTS_CONFIG, that's the mechanism to keep the step simple and generic. The step offers also an additional customization mechanism (dev-scripts-additional-config) for very specific needs, but the preferred way (also for the maintenance) remains DEVSCRIPTS_CONFIG

Reply:
@andfasano So the issue is with translating the image url, not the actual mechanism for adding it to dev-scripts.
I.e. see the setup ob [log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/42043/rehearse-42043-pull-ci-openshift-appliance-master-e2e-sno-dualstack-dhcp/1688832932581478400/artifacts/e2e-sno-dualstack-dhcp/baremetalds-devscripts-setup/build-log.txt):
```
export APPLIANCE_IMAGE=registry.build05.ci.openshift.org/ci-op-hn18sxx2/pipeline@sha256:7a4c03459d76ee47f7ec0657abe1648855d6ddbe28673563d0376606c192d5d6
```
I.e. the value can't be hard-coded. Any suggestion for adding it without using dependencies?

Reply:
I don't think there's a way to do that without the dependencies mechanism (which is the main method in prow to pass built/imported images)

Reply:
After discussing with @andfasano, added a condition to 'baremetalds-devscripts-setup-commands.sh' for ensuring that the APPLIANCE_IMAGE var is added only for appliance e2e jobs.

Reply:
Unfortunately the dependency definition it's possible only at the step level (while the override can be set elsewhere) and yes, there's no other way currently available to inject a pullspec. The current approach at least avoids setting the APPLIANCE_IMAGE var when not required